Automattic/jetpack
# Social: Add new connection modal

Fixes https://github.com/Automattic/jetpack-reach/issues/296

This changes adds the "Add new connection" modal to the plugin, which will be used with the connection management component that's being added in #37120

## Proposed changes:

* Added the new modal
* The modal has two pages, first one is the service selector, the second one is the pre-connection page, which is different for each service
* Text is copied from `https://cloud.jetpack.com/jetpack-social/<SITE>`

### Other information:


## Jetpack product discussion


https://github.com/Automattic/jetpack-reach/issues/296
## Does this pull request change what data or activity we track or use?


No.
## Testing instructions:






NOTE: The colors of the social icons is black, because the CSS variables from Calypso are not loaded properly. This will be fixed in this or a following change.

* Go to the Social admin page, with `useAdminUiV1` defined.
* Click the `Add new connection` button in the connection list component. The modal window should open.
* Check the copy, then click on the next button beside one of the connections.
* Check that the page makes sense. The connect button is not hooked up in the PR.
* Go back to the main screen and do the same with all connections.

Service selector page:
![CleanShot 2024-05-07 at 12 31 03 png](https://github.com/Automattic/jetpack/assets/36671565/0e934329-c770-4c92-8607-0adcb8551b71)
Examples of pre-connection pages:
![CleanShot 2024-05-06 at 16 57 42 png](https://github.com/Automattic/jetpack/assets/36671565/8f9d68ad-247a-4697-b977-0df23e63f4e1)
![CleanShot 2024-05-06 at 16 57 47 png](https://github.com/Automattic/jetpack/assets/36671565/75b4965f-0a57-40c1-b18e-3b7dcc398dfe)
